File: Beecrowd/2533.cpp
#include <iostream>
#include <iomanip>
using namespace std;

int main() {
    int M;

    while (cin >> M) {
        double notas, cargaHoraria;
        double somaPonderada = 0, totalCargaHoraria = 0;

        for (int i = 0; i < M; i++) {
            cin >> notas >> cargaHoraria;
            somaPonderada += notas * cargaHoraria; 
            totalCargaHoraria += cargaHoraria; 
        }

        double IRA = (somaPonderada / (totalCargaHoraria * 100)); 

        cout << fixed << setprecision(4) << IRA << endl;
    }

    return 0;
}

File: Beecrowd/2533.py
while True:
    try:
        M = int(input())
    except EOFError:
        break

    soma = 0
    totalCargaHoraria = 0

    for _ in range(M):
        notas, cargaHoraria = map(float, input().split()) 
        soma += notas * cargaHoraria
        totalCargaHoraria += cargaHoraria

    IRA = soma / (totalCargaHoraria * 100)
    print(f"{IRA:.4f}") 
